app/moderation-algo/main.py

```diff
@@ -0,0 +1,79 @@
+    @Return: {'issue': issue, 'topics': topics}"""
+    # Make sure Google API is initialized
+    model = 'gemini-pro'
+    gemini.InitGoogleGemini()
+    issue = GetIssue(model, reading_materials)
+    topics = GetTopics(model, reading_materials)
+    return {'issue': issue, 'topics': topics}
+
+def ProcessQuestion(issue: str, topics: dict, question: str) -> dict:
+    """Process a question through the algorithm and return the results from each stage as a dict.
+    @Param: issue = Broad issue that the town hall is about (Ex: 'modernizing congress')
+    @Param: question = User question or comment to process
+    @Return: results = {'substantive': bool, 'offensive': bool, 'relevant': bool, 'topics': {'topic0': bool, 'topic1': bool, ...}}"""
+    # NOTE: Make sure Google API is initialized (Ex: set env variable GOOGLE_APPLICATION_CREDENTIALS to filepath with API key)
+    model = 'gemini-pro'
+
+    # Process the question through the algo
+    response = {'substantive': False, 'offensive': False, 'relevant': False, 'topics': {}}
+
+    # Check for substantiveness
+    response['substantive'] = IsSubstantive(model, question)
+    if(not response['substantive']):
+        return response
+
+    # Check for offensiveness
+    response['offensive'] = IsOffensive(question)
+    if(response['offensive']):
+        return response
+
+    # Check for relevancy
+    response['relevant'] = IsRelevant(model, question, issue)
+
+    # Classify into topics
+    for topic, description in topics.items():
+        response['topics'][topic] = DoesQuestionFitCategory(model, question, topic, description)
+
+    return response
+
+@app.route('/', methods=['POST'])
+def HandleUserInput():
+    # Check if the request contains JSON data
+    if request.is_json:
+        print(request)
+        # If reading materials are provided, return their discussed issue and topics / descriptions
+        reading_materials = request.get_json().get("reading_materials")
+        if(reading_materials):
+            return jsonify(GetIssueTopics(reading_materials)), 200 # HTTP success
+
+        # Otherwise, 
+        issue = request.get_json().get("issue") # Get overall issue (Ex: 'modernizing congress')
+        topics = request.get_json().get("topics") # All possible topics and their description {'topic': 'description', ...}
+        question = request.get_json().get("question") # Get the user question/comment
+        if(question and issue and topics):
+            # Process user question through the algorithm and return output
+            return jsonify(ProcessQuestion(issue, topics, question)), 200 # HTTP success
+        else:
+            return jsonify({'ERROR': 'Missing field(s) in request data'}), 400 # HTTP bad request
+    else:
+        return jsonify({'ERROR': 'Request must be in JSON format'}), 415 # HTTP unsupported media type
+
+if __name__ == '__main__':
+    InitPerspectiveAPI()
+    app.run(debug=True, host='0.0.0.0', port=9989)
```

Comment:
The warning `WARNING: This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.` shows when running the API, could you add a check so it runs a production server when `NODE_ENV=production`?